src/data_member.rs

```diff
@@ -72,6 +72,108 @@ impl<T: Sized + Copy> DataMember<T> {
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Create a new `DataMember` from a [`ProcessHandle`], pointing to the memory address in the
+    /// remote process. Equivalent to `new_offset(handle, vec![addr])`. You must
+    /// remember to call [`try_into_process_handle`] on a [`Pid`] as sometimes the `Pid` can have
+    /// the same backing type as a [`ProcessHandle`], resulting in an error.
+    ///
+    /// [`try_into_process_handle`]: trait.TryIntoProcessHandle.html#tymethod.try_into_process_handle
+    /// [`ProcessHandle`]: type.ProcessHandle.html
+    /// [`Pid`]: type.Pid.html
+    #[must_use]
+    pub fn new_addr(handle: ProcessHandle, addr: usize) -> Self {
+        Self {
+            offsets: vec![addr],
+            process: handle,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Create a new `DataMember` from a [`ProcessHandle`], pointing to the memory address in the
+    /// process, then following a bunch of pointers and offsets, which may be negative.
+    /// If you use CheatEngine and get a pointer of form "MyModule.dll + 0x12345678", plus a bunch
+    /// of offsets, then you want to put the base address of the module as `addr`, `0x12345678` as
+    /// the first offset, then any further offsets etc.
+    /// This function is merely a convenience function, and is equivalent to
+    /// `new_offset_relative(handle, vec![addr as isize, offsets[0], offsets[1], ...])`.
+    /// You must
+    /// remember to call [`try_into_process_handle`] on a [`Pid`] as sometimes the `Pid` can have
+    /// the same backing type as a [`ProcessHandle`], resulting in an error.
+    ///
+    /// [`try_into_process_handle`]: trait.TryIntoProcessHandle.html#tymethod.try_into_process_handle
+    /// [`ProcessHandle`]: type.ProcessHandle.html
+    /// [`Pid`]: type.Pid.html
+    #[must_use]
+    #[allow(clippy::doc_markdown)]
+    #[allow(clippy::cast_sign_loss)]
+    pub fn new_addr_offset(handle: ProcessHandle, addr: usize, offsets: Vec<isize>) -> Self {
+        let mut vec = vec![addr];
+        // Yes, we are casting to usize. This will not touch any bits, and due to 2s complement,
+        // we still get the correct result when adding offsets.
+        vec.extend(offsets.into_iter().map(|x| x as usize));
+        Self {
+            offsets: vec,
+            process: handle,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Creates a new `DataMember` by appending some more offets. Useful when you have a data
+    /// structure, and want to refer to multiple fields in it, or use it as a starting point
+    /// for chasing down more pointers.
+    /// Since the pointed-to data type might have changed, this function is generic. It is your
+    /// responsibility to make sure you know what you point to.
+    #[allow(clippy::cast_sign_loss)]
+    #[must_use]
+    pub fn extend<TNew>(&self, more_offsets: Vec<isize>) -> DataMember<TNew> {
```

Comment:
Can these functions be put on the `Memory` trait instead? I think that they would also be useful on `LocalMember`.